crates/engine/tree/src/tree/cached_state.rs

```diff
@@ -266,23 +388,250 @@ pub(crate) struct ProviderCacheBuilder {
 impl ProviderCacheBuilder {
     /// Build a [`ProviderCaches`] struct, so that provider caches can be easily cloned.
     pub(crate) fn build_caches(self) -> ProviderCaches {
-        ProviderCaches {
-            code_cache: CacheBuilder::new(self.code_cache_size)
-                .build_with_hasher(DefaultHashBuilder::default()),
-            storage_cache: CacheBuilder::new(self.storage_cache_size)
-                .build_with_hasher(DefaultHashBuilder::default()),
-            account_cache: CacheBuilder::new(self.account_cache_size)
-                .build_with_hasher(DefaultHashBuilder::default()),
-        }
+        const STORAGE_MAX_WEIGHT: u64 = 8 * 1024 * 1024 * 1024; // 8Gb
```

Comment:
that means we're increasing the hardware requirements to 9GB minimum, right? should we instead make it dynamic depending on the machine, or at least lower the storage max weight to 2-4GB?

Reply:
> or at least lower the storage max weight to 2-4GB

finally updated to use at most 4GB which showed improved results in the benchmarks compared to the baseline, also now there's a single value that determines the sizes of the 3 caches